modules/monitoring-resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus.adoc

```diff
@@ -0,0 +1,94 @@
+= Resolving the KubePersistentVolumeFillingUp alert firing for Prometheus
+
+As a cluster administrator, you can resolve the `KubePersistentVolumeFillingUp` alert being triggered for Prometheus. 
+
+The critical alert fires when a persistent volume (PV) claimed by a `prometheus-k8s-*` pod in the `openshift-monitoring` project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.
+
+[NOTE]
+====
+There are two `KubePersistentVolumeFillingUp` alerts:
+
+* *Critical alert*:  The alert with the `severity="critical"` label is triggered when the mounted PV has less than 3% of its total space remaining.
+* *Warning alert*: The alert with the `severity="warning"` label is triggered when the mounted PV has less than 15% of its total space remaining and is expected to fill up within four days.
+====
+
+To address this issue, you can remove Prometheus time series database (TSDB) blocks to create more space for the PV.
+
+.Prerequisites
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `cluster-admin` cluster role.
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `dedicated-admin` role.
+endif::openshift-dedicated,openshift-rosa[]
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. List the size of all TSDB blocks, sorted from oldest to newest:
++
+[source,terminal]
+----
+$ oc debug <prometheus_k8s_pod_name> -n openshift-monitoring \// <1>
+-c prometheus --image=$(oc get po -n openshift-monitoring <prometheus_k8s_pod_name> \// <1>
+-o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}') \
+-- sh -c 'cd /prometheus/;du -hs $(ls -dt */ | grep -Eo "[0-9|A-Z]{26}")'
+----
+<1> Replace `<prometheus_k8s_pod_name>` with the pod mentioned in the `KubePersistentVolumeFillingUp` alert description.
++
+.Example output
+[source,terminal]
+----
+308M    01HVKMPKQWZYWS8WVDAYQHNMW6
+52M     01HVK64DTDA81799TBR9QDECEZ
+102M    01HVK64DS7TRZRWF2756KHST5X
+140M    01HVJS59K11FBVAPVY57K88Z11
+90M     01HVH2A5Z58SKT810EM6B9AT50
+152M    01HV8ZDVQMX41MKCN84S32RRZ1
+354M    01HV6Q2N26BK63G4RYTST71FBF
+156M    01HV664H9J9Z1FTZD73RD1563E
+216M    01HTHXB60A7F239HN7S2TENPNS
+104M    01HTHMGRXGS0WXA3WATRXHR36B
+----
+
+. Identify which and how many blocks could be removed. For example, to remove the three oldest Prometheus TSDB blocks, run the following command:
++
+[source,terminal]
+----
+$ oc debug <prometheus_k8s_pod_name> -n openshift-monitoring \// <1>
+-c prometheus --image=$(oc get po -n openshift-monitoring <prometheus_k8s_pod_name> \// <1>
+-o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}') \
+-- sh -c 'ls -latr /prometheus/ | egrep -o "[0-9|A-Z]{26}" | head -3 | \
+while read BLOCK; do rm -r /prometheus/$BLOCK; done'
+----
+<1> Replace `<prometheus_k8s_pod_name>` with the pod mentioned in the `KubePersistentVolumeFillingUp` alert description.
```

Comment:
Not super necessary, but since this is only an example, you could replace the input values with an example name if you wanted. Not sure what the correct syntax for a pod name is, though, so it might not be worth the effort if looking that up would take a long time:
```suggestion
$ oc debug prometheus-k8s-example-pod -n openshift-monitoring \
-c prometheus --image=$(oc get po -n openshift-monitoring prometheus-k8s-example-pod \
-o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}') \
-- sh -c 'ls -latr /prometheus/ | egrep -o "[0-9|A-Z]{26}" | head -3 | \
while read BLOCK; do rm -r /prometheus/$BLOCK; done'
----
```

I'll leave that up to you.

Reply:
Yes, you are right, maybe giving the complete example would be better here :+1: